server/subscribe_test.go

```diff
@@ -17,6 +17,74 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestListChannelSubscriptions(t *testing.T) {
+	p := &Plugin{}
+	p.updateConfig(func(conf *config) {
+		conf.Secret = "somesecret"
+	})
+
+	for name, tc := range map[string]struct {
+		Subs            *Subscriptions
+		Expected string
+	}{
+		"test 1": {
+			Subs: withExistingChannelSubscriptions([]ChannelSubscription{
+				ChannelSubscription{
+					Id:        model.NewId(),
+					ChannelId: "channel1",
+					Name: "Sub Name X",
+					Filters: SubscriptionFilters{
+						Projects: NewStringSet("PROJ"),
+					},
+				},
+			}),
+			Expected: `~channel-1-name (1):
+* PROJ - Sub Name X`,
+		},
```

Comment:
I wasn't able to unit test much because the order of data is unknown, since the kv data structures are all maps. So I just tested having one subscription.